samples/volumes/example5.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+
+
+@dsl.pipeline(
+    name="Example 5",
+    description="The fifth example of the design doc."
+)
+def example5(rok_url):
+    vol1 = dsl.PipelineVolume(
+        name="vol1",
+        size="1Gi",
+        annotations={"rok/origin": rok_url},
+        mode=dsl.VOLUME_MODE_RWM
+    )
+
+    step1 = dsl.ContainerOp(
+        name="step1_concat",
+        image="library/bash:4.4.23",
+        command=["sh", "-c"],
+        arguments=["cat /data/file*| gzip -c >/data/full.gz"],
+        volumes={"/data": vol1}
```

Comment:
It makes sense to me to support volumes in ContainerOp's constructor, but it's better to make it support a generic k8s volume spec here.  PipelineVolume still have value to exist to hide the complexity of k8s sdk code and it may keep additional data other than k8s Volume like subPath to facilitate mounting inputs and output folders from a single volume.

Reply:
@hongye-sun 
> It makes sense to me to support volumes in ContainerOp's constructor, but it's better to make it support a generic k8s volume spec here. PipelineVolume still have value to exist to hide the complexity of k8s sdk code and it may keep additional data other than k8s Volume like subPath to facilitate mounting inputs and output folders from a single volume.

ACK. We must simplify the way `ContainerOp` instsances can consume arbitrary `V1Volume` instances. If we make `PipelineVolume` a descendant of `V1Volume` then things will work automagically, whether we create `k8s.V1Volume` instances manually, or we depend on a previous task in the pipeline emitting a `PipelineVolume` instance.